plugins/modules/register_cluster.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import, division, print_function
+
+import json
+import os
+import requests
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: register_cluster
+
+short_description: Register AssistedInstaller cluster
+
+version_added: "1.0.0"
+
+description: Register an AssistedInstaller Cluster
+
+options:
+    data:
+        description: JSON data for AssistedInstaller
+        type: dict
+        requried: true
+    NOTE: AI_API_TOKEN and AI_PULL_SECRET env variables must be set
```

Comment:
Is the reason for AI_PULL_SECRET to be an env variable because its a secret that we do not want to accidentally leave in a playbook and have it exposed, right?
cc: @dkosteck @tonyskapunk

Reply:
That's one reason, yes, similar to how AI_API_TOKEN is used